File: Baekjoon/Class2/36.1018 체스판 다시 칠하기.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

// 나머지 연산
const countPaint = (board, row, col) => { // 색칠해야 하는 칸 세기
    let count = 0;
    for(let i = row; i < row + 8; i++)
        for(let j = col; j < col + 8; j++)
            if((i + j) % 2 === 0 && board[i][j] === "B") // 잘못 색칠된 검정
                count++;
            else if((i + j) % 2 === 1 && board[i][j] === "W") // 잘못 색칠된 하양
                count++;
    return count;
}

/* 체스판 생성
const countPaint = (board, row, col) => { // 색칠해야 하는 칸 세기
    let rowFirst = "W" // 흰색으로 시작하는 체스 보드
    let count = 0;
    for(let i = 0; i < 8; i++) {
        cur = rowFirst;
        for(let j = 0; j <  8; j++) {
            if(board[i + row][j + col] != cur)
                count++;
            cur = (cur == 'W') ? 'B' : 'W';
        }
        rowFirst = (rowFirst == 'W') ? 'B' : 'W';
    }
    return count;
}
*/

/* 체스판 정의
const CHESS_BOARD = [ // 흰색으로 시작하는 체스 보드
    "WBWBWBWB",
    "BWBWBWBW",
    "WBWBWBWB",
    "BWBWBWBW",
    "WBWBWBWB",
    "BWBWBWBW",
    "WBWBWBWB",
    "BWBWBWBW"
]
const countPaint = (board, row, col) => { // 색칠해야 하는 칸 세기
    let count = 0;
    for(let i = 0; i < 8; i++)
        for(let j = 0; j <  8; j++)
            if(board[i + row][j + col] != CHESS_BOARD[i][j])
                count++;
    return count;
}
*/

const [n, m] = input[0].split(" ").map(Number);
const board = input.slice(1);

let minCount = 64;
for(let row = 0; row <= n - 8; row++)
    for(let col = 0; col <= m - 8; col++) {
        let count = countPaint(board, row, col);
        // 흰색으로 시작할 때 색칠할 칸, 반전한 경우, 이전 최소값 중 최소값 선택
        minCount = Math.min(count, 64 - count, minCount);
    }
console.log(minCount);

File: Baekjoon/Class2/36.1018 체스판 다시 칠하기.py
import sys
input = sys.stdin.readline

#나머지 연산
def count_paint(board, row, col): #색칠해야 하는 칸 세기
    count = 0
    for i in range(row, row + 8):
        for j in range(col, col + 8):
            if (i + j) % 2 == 0 and board[i][j] == "B": #잘못 색칠된 검정
                count += 1
            elif (i + j) % 2 == 1 and board[i][j] == "W": #잘못 색칠된 하양
                count += 1
    return count

''' 체스판 생성
def count_paint(board, row, col): #색칠해야 하는 칸 세기
    row_first = "W" #흰색으로 시작하는 체스 보드
    count = 0
    for i in range(8):
        cur = row_first
        for j in range(8):
            if board[i + row][j + col] != cur:
                count += 1
            cur = "B" if cur == "W" else "W"
        row_first = "B" if row_first == "W" else "W"
    return count
'''

''' 체스판 선언
CHESS_BOARD = [ #흰색으로 시작하는 체스 보드
    "WBWBWBWB",
    "BWBWBWBW",
    "WBWBWBWB",
    "BWBWBWBW",
    "WBWBWBWB",
    "BWBWBWBW",
    "WBWBWBWB",
    "BWBWBWBW"
]
def count_paint(board, row, col): #색칠해야 하는 칸 세기
    count = 0
    for i in range(8):
        for j in range(8):
            if board[i + row][j + col] != CHESS_BOARD[i][j]:
                count += 1
    return count
'''

def main():
    n, m = map(int, input().split())
    board = [input() for _ in range(n)]
    min_count = 64
    for row in range(n - 7):
        for col in range(m - 7):
            count = count_paint(board, row, col)
            #흰색으로 시작할 때 색칠할 칸, 반전한 경우, 이전 최소값 중 최소값 선택
            min_count = min(count, 64 - count, min_count)
    print(min_count)

main()